File: py-lambda-worker/app/worker.py
import asyncio
import os

from temporalio.common import WorkerDeploymentVersion
from temporalio.worker import Worker, WorkerDeploymentConfig

from .activities import greet
from .connection import connect
from .workflows import GreetingWorkflow


async def main() -> None:
    task_queue = os.environ.get("TEMPORAL_TASK_QUEUE", "py-lambda-worker-queue")
    deployment_name = os.environ.get("TEMPORAL_DEPLOYMENT_NAME", "py-lambda-worker")
    build_id = os.environ.get("TEMPORAL_BUILD_ID", "v1.0")

    client = await connect()
    worker = Worker(
        client,
        task_queue=task_queue,
        workflows=[GreetingWorkflow],
        activities=[greet],
        deployment_config=WorkerDeploymentConfig(
            version=WorkerDeploymentVersion(
                deployment_name=deployment_name,
                build_id=build_id,
            ),
            use_worker_versioning=True,
        ),
    )
    print(
        f"Starting worker on task queue: {task_queue}"
        f" deployment={deployment_name} build_id={build_id}"
    )
    await worker.run()


if __name__ == "__main__":
    asyncio.run(main())

File: cloud-run/src/worker.ts
import { readFileSync } from "node:fs";
import { join } from "node:path";
import { NativeConnection, Worker } from "@temporalio/worker";
import * as activities from "./activities";
import { loadConfig } from "./config";
import { installRuntime } from "./runtime";

async function run(): Promise<void> {
  const config = loadConfig();
  installRuntime(config);

  const connection = await NativeConnection.connect({
    address: config.address,
    ...(config.apiKey ? { apiKey: config.apiKey, tls: true } : {}),
  });

  const worker = await Worker.create({
    connection,
    namespace: config.namespace,
    taskQueue: config.taskQueue,
    workflowBundle: {
      code: readFileSync(join(__dirname, "workflow-bundle.js"), "utf8"),
    },
    activities,
    ...(config.useVersioning
      ? {
          workerDeploymentOptions: {
            version: {
              deploymentName: config.deploymentName,
              buildId: config.buildId,
            },
            useWorkerVersioning: true,
            defaultVersioningBehavior: "PINNED" as const,
          },
        }
      : {}),
  });

  console.log(
    `Starting worker: namespace=${config.namespace} taskQueue=${config.taskQueue}` +
      (config.useVersioning
        ? ` deployment=${config.deploymentName} buildId=${config.buildId}`
        : " versioning=off"),
  );

  try {
    await worker.run();
  } finally {
    await connection.close();
  }
}

run().catch((error: unknown) => {
  console.error(error);
  process.exitCode = 1;
});
